src/vm/peimage.inl

```diff
@@ -481,8 +481,19 @@ inline BOOL PEImage::HasNativeHeader()
         return GetLoadedLayout()->HasNativeHeader();
     else
     {
-        PEImageLayoutHolder pLayout(GetLayout(PEImageLayout::LAYOUT_ANY,LAYOUT_CREATEIFNEEDED));
```

Comment:
Thank you @sdmaclea. I would suggest to handle it closer to the intended place where we really want to handle, while keeping the other use-cases unaffected.

Even at the proposed location, I would suggest enabling the try/catch only for Arm64 so that we are intentional about where this fallback is going to work.